### PR TITLE
support loc config conversion

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3447

File: src/docfx/build/metadata/ContributionInfo.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ public DateTime GetUpdatedAt(Document document, List<GitCommit> fileCommits)
                 return default;
 
             var branch = repo.Branch ?? "master";
-            var editRepo = document.Docset.Config.Contribution.Repository ?? $"{repo.Owner}/{repo.Name}";
+            var editRepo = LocConfigConversion.GetEditRepository(document.Docset.Config.Contribution.Repository, document.Docset.Locale) ?? $"{repo.Owner}/{repo.Name}";
```
Comment (live1206) [inline src/docfx/build/metadata/ContributionInfo.cs]: naming consistent, shall we use loc or locale?

Comment (live1206) [reply src/docfx/build/metadata/ContributionInfo.cs]: should be convention?

Comment (OsmondJiang) [reply src/docfx/build/metadata/ContributionInfo.cs]: loc means localization, not locale

File: test/docfx.Test/common/TestUtility.cs [hunk 2]
```diff
@@ -53,6 +53,13 @@ public static void VerifyJsonContainEquals(JToken expected, JToken actual, strin
                     Assert.Equal(TestUtility.NormalizeHtml(expectedHtml), TestUtility.NormalizeHtml(actualHtml));
                 }
                 else
+                if (expectedValue is string expectedStr && actualValue is string actualStr &&
+                    expectedStr.StartsWith("*") && expectedStr.EndsWith("*"))
```
Comment (live1206) [inline test/docfx.Test/common/TestUtility.cs]: where is this being used?

File: src/docfx/config/LocConfigurationConvention.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            {
+                return repository;
+            }
+
+            var repoName = repository.Split(new[] { '/', '\\' }).Last();
+            var match = s_repoNameWithLocale.Match(repoName);
+            if (match.Success && match.Groups.Count == 2 && !string.IsNullOrEmpty(match.Groups[1].Value))
+            {
+                var originLocale = match.Groups[1].Value;
+                return repository.Replace(originLocale, $".{locale}");
```
Comment (superyyrrzz) [inline src/docfx/config/LocConfigurationConvention.cs]: >repository.Replace [](start = 23, length = 18)

use regex.replace? considier if a repo named `azure-betwen-us.en-us`

**Pull request merged**